batch/batch/driver/job.py

```diff
@@ -355,22 +356,22 @@ async def job_config(app, record, attempt_id):
 
     userdata = json.loads(record['userdata'])
 
-    secrets = job_spec.get('secrets', [])
-    k8s_secrets = await asyncio.gather(
-        *[k8s_cache.read_secret(secret['name'], secret['namespace']) for secret in secrets]
-    )
-
     gsa_key = None
+    if not os.environ.get('HAIL_TERRA'):
```

Comment:
same question here: should it just be an assertion that there are no secrets when we're in hail terra?